src/providers/ldap/sdap.c

```diff
@@ -1180,6 +1180,7 @@ sdap_create_search_base(TALLOC_CTX *mem_ctx,
         goto done;
     }
 
+    base->ldb_basedn = ldn;
```

Comment:
Geez. You are right. Although I didn't experience the crash, the code confirms what you say.

I'm working on getting and storing the long-living `ldb_context`s in the structure, but retrieving it starts to be like a tree with many branches. The problem of this is that it will be quite complex (error-prone) to guarantee none of them is freed at some time while keeping the reference on this structure.

Reply:
It's fixed.

I kept it in a separate commit so that it can be easily identified, reviewed and rolled back if needed. If we agree on this, later I will merge this last commit to the original.

@thalman This is a significant change you might want to take a look at.